packages/app/src/cli/services/dev/ui/components/Dev.tsx

```diff
@@ -136,6 +142,11 @@ const Dev: FunctionComponent<DevProps> = ({
     return () => clearInterval(devSessionPollingInterval.current)
   }, [devSessionStatus])
 
+  useEffect(() => {
+    const newMessage = `Preview URL: ${appPreviewURL}${graphiqlUrl ? `\nGraphiQL URL: ${localhostGraphiqlUrl}` : ''}`
+    if (!isShuttingDown) setStatusMessage(newMessage)
+  }, [appPreviewURL])
```

Comment:
I'm happy to leave this out of scope in this PR. But the `useEffect` calls here have some incorrect dependency arrays. I think that's also leading to using state for things that actually could just be values produced at render time. Worth a clean-up, and enabling React's recommended linting rules, once this has landed imo. Not fixing may lead to strange bugs.

e.g. this one should be `[appPreviewURL, graphiqlUrl, localhostGraphiqlUrl, isShuttingDown]`; probably the effect & the status message state aren't needed.

Reply:
Agree we should improve the state and useEffect in general in this file, but for those specific value, `graphiqlUrl` and `localhostGraphiqlUrl` are not state (they can't change at runtime)

And `isShuttingDown` is not state, but it could be